<?xml version="1.0"?>

<launch>

    <arg name="autorun" default="false" />

    <node required="false" pkg="cartiber" type="cartiber" name="cartiber" output="screen"
          launch-prefix="bash -c 'sleep 3; $0 $@' ">

        <param name="priormap_file"  value="/home/tmn/ros_ws/dev_ws/src/cartiber/scripts/priormap.pcd" />

        <!-- <param name="lidar_bag_file" value="/home/tmn/ros_ws/dev_ws/src/cartiber/scripts/cloud_avia_mid_no_offset.bag" /> -->
        <param name="lidar_bag_file" value="/home/tmn/ros_ws/dev_ws/src/cartiber/scripts/cloud_avia_mid.bag" />
        
        <param name="MAX_CLOUDS"     value="100"/>
        <param name="pc_topics"      value="['/lidar_1/points']" type="yaml" />
        <param name="lidar_type"     value="['ouster', 'livox']" type="yaml" />
        <param name="xyzypr_W_L0"    value="[
                                             0, 0, 0.5, -135, 00, 0,
                                             0, 0, 0.7,  045, 45, 0
                                            ]"                   type="yaml"/>
        <!-- EKF params -->
        <param name="leaf_size"      value="0.15" />
        <param name="UW_NOISE"       value="10.0" />
        <param name="UV_NOISE"       value="10.0" />
        <param name="DEBUG_STEPS"    value="0"    />
        <param name="MAX_ITER"       value="5"    />
        <param name="dXmax"          value="0.2"  />
        
        <!-- GP MAP optimization params -->
        <param name="deltaT"         value="0.043"/>
        <param name="WINDOW_SIZE"    value="3"    />
        <param name="fixed_start"    value="0.001" />
        <param name="fixed_end"      value="-0.001" />

        <param name="lidar_ds_rate"  value="10"    />
        <param name="lidar_weight"   value="1.0"  />
        
        <!-- Pose prior factor -->
        <param name="ppSigmaR"       value="-1.0" />
        <param name="ppSigmaP"       value="-1.0" />
        
        <!-- Motion prior factors -->
        <param name="mpSigmaR"       value="10"  />
        <param name="mpSigmaP"       value="10"  />
        <param name="DK"             value="2"    />
        <param name="tshift"         value="0.025"/>

        <!-- Smoothness factors -->
        <param name="smSigmaR"       value="-10" />
        <param name="smSigmaP"       value="-10" />

        <!-- Loss function threshold -->
        <param name="ld_loss_thres"  value="1.0"   />
        <param name="pp_loss_thres"  value="-1.0"  />
        <param name="mp_loss_thres"  value="-1.0"  />
        <param name="sm_loss_thres"  value="-1.0"  />

    </node>

    <!-- Visualize -->
    <node pkg="rviz" type="rviz" name="rviz_odom" output="log" required="true"
        args="-d $(find cartiber)/launch/cartiber.rviz" />

</launch>